packages/@aws-cdk/aws-codepipeline/lib/stage.ts

```diff
@@ -67,35 +55,32 @@ export interface StageProps extends CommonStageProps {
  *
  * @example
  *   // add a Stage to a Pipeline
- *   new Stage(this, 'MyStage', {
- *     pipeline: myPipeline,
+ *   pipeline.addStage(new Stage('MyStage', {
+ *     actions: [
+ *       // ...
+ *     ],
  *   });
  */
-export class Stage extends cdk.Construct implements cpapi.IStage, cpapi.IInternalStage {
-  /**
-   * The Pipeline this Stage is a part of.
-   */
-  public readonly pipeline: cpapi.IPipeline;
-  public readonly name: string;
+export class Stage implements cpapi.IStage {
```

Comment:
Let's explore the idea of not exposing this as a public API and just expose the interface

Reply:
Done.